From e8b221cdc37e2a7a15395c406cb1792d6e84c538 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Sun, 30 May 2021 21:52:36 +0300
Subject: [PATCH 249/302] media: i2c: imx296: Replace gain control with
 analogue gain

libcamera expects V4L2_CID_ANALOGUE_GAIN, not V4L2_CID_GAIN.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
---
 drivers/media/i2c/imx296.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/imx296.c b/drivers/media/i2c/imx296.c
index 6b1b31216016..70988dd9ec13 100644
--- a/drivers/media/i2c/imx296.c
+++ b/drivers/media/i2c/imx296.c
@@ -278,7 +278,7 @@ static int imx296_s_ctrl(struct v4l2_ctrl *ctrl)
 		imx296_write(imx, IMX296_SHS1, vmax - ctrl->val, &ret);
 		break;
 
-	case V4L2_CID_GAIN:
+	case V4L2_CID_ANALOGUE_GAIN:
 		imx296_write(imx, IMX296_GAIN, ctrl->val, &ret);
 		break;
 
@@ -341,7 +341,7 @@ static int imx296_ctrls_init(struct imx296 *imx)
 	v4l2_ctrl_new_std(&imx->ctrls, &imx296_ctrl_ops,
 			  V4L2_CID_EXPOSURE, 1, 1048575, 1, 1104);
 	v4l2_ctrl_new_std(&imx->ctrls, &imx296_ctrl_ops,
-			  V4L2_CID_GAIN, IMX296_GAIN_MIN,
+			  V4L2_CID_ANALOGUE_GAIN, IMX296_GAIN_MIN,
 			  IMX296_GAIN_MAX, 1, IMX296_GAIN_MIN);
 
 	/*
-- 
Regards,

Laurent Pinchart

